ucsb-cs156-f23/proj-happycows-f23-5pm-1
# Vl Last Date, As an admin I can control dates for game play

## Overview

In this PR I added a lastDate value for Commons. This means I changed values in the Commons.java and CommonsController files. Furthermore in the frontend I edited the Commons form and table, admin create and edit pages. The relevant tests were added for all these changes.

## Screenshots (Optional)

![image](https://github.com/ucsb-cs156-f23/proj-happycows-f23-5pm-1/assets/87053587/7d983e04-24cb-43cd-8b71-941012fdf135)


## Future Possibilities (Optional)

Some possible points could be:
- For the automatic jobs that manipulate cow health, milk the cows, etc. we should restrict which commons these are done on to only the commons that have games in progress. There is no reason to milk cows, update health, produce instructor reports etc. for a game that's already over. It's important that these activities STOP at some point, or else we'll be filling up the database with useless rows indefinitely. That's how you crash a server.

- A clean start/end date will likely make future graphs and visualizations easier to make.

- We may want to restrict users from joining a commons before the game start date so that the instructor can set things up in advance, but students all start at the same time.

## Validation (Optional)

1. Download this branch.
2. Run on local host
3. Test Admin create and edit commons forms
4. Make sure last date is saved, by checking swagger or the table

## Tests


## Linked Issues

Closes #15
